src/server/task_manager/task_manager.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Joi from 'joi';
+import {
+  Logger,
+  TaskDefinition,
+  TaskDictionary,
+  TaskManager,
+  TaskPool,
+  TaskStore,
+  validateTaskDefinition,
+} from './task_pool';
+import { SanitizedTaskDefinition } from './task_pool/task';
+
+function taskManagerFactory(kbnServer: any, server: any, config: any) {
+  let taskManager: TaskManager;
```

Comment:
the function being returned here will have to move to a `start` function that kicks off after all the plugins init